src/libs/ReportUtils.ts

```diff
@@ -20,174 +30,448 @@ import Navigation from './Navigation/Navigation';
+} & Report;
+
+type OnyxDataTaskAssigneeChat = {
+    optimisticData: OnyxUpdate[];
+    successData: OnyxUpdate[];
+    failureData: OnyxUpdate[];
+    optimisticAssigneeAddComment?: OptimisticReportAction;
+    optimisticChatCreatedReportAction?: OptimisticCreatedReportAction;
+};
+
+let currentUserEmail: string | undefined;
+let currentUserAccountID: number | undefined;
+let isAnonymousUser = false;
 
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
+    callback: (value) => {
         // When signed out, val is undefined
-        if (!val) {
+        if (!value) {
             return;
         }
 
-        currentUserEmail = val.email;
-        currentUserAccountID = val.accountID;
-        isAnonymousUser = val.authTokenType === 'anonymousAccount';
+        currentUserEmail = value.email;
+        currentUserAccountID = value.accountID;
+        isAnonymousUser = value.authTokenType === 'anonymousAccount';
     },
 });
 
-let allPersonalDetails;
-let currentUserPersonalDetails;
+let allPersonalDetails: OnyxCollection<PersonalDetails>;
+let currentUserPersonalDetails: OnyxEntry<PersonalDetails>;
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-    callback: (val) => {
-        currentUserPersonalDetails = lodashGet(val, currentUserAccountID, {});
-        allPersonalDetails = val || {};
+    callback: (value) => {
+        currentUserPersonalDetails = value?.[currentUserAccountID ?? -1] ?? null;
+        allPersonalDetails = value ?? {};
     },
 });
 
-let allReports;
+let allReports: OnyxCollection<Report>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
-    callback: (val) => (allReports = val),
+    callback: (value) => (allReports = value),
 });
 
-let doesDomainHaveApprovedAccountant;
+let doesDomainHaveApprovedAccountant = false;
 Onyx.connect({
     key: ONYXKEYS.ACCOUNT,
-    waitForCollectionCallback: true,
-    callback: (val) => (doesDomainHaveApprovedAccountant = lodashGet(val, 'doesDomainHaveApprovedAccountant', false)),
+    callback: (value) => (doesDomainHaveApprovedAccountant = value?.doesDomainHaveApprovedAccountant ?? false),
 });
 
-let allPolicies;
+let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     waitForCollectionCallback: true,
-    callback: (val) => (allPolicies = val),
+    callback: (value) => (allPolicies = value),
 });
 
-let loginList;
+let loginList: OnyxEntry<Login>;
 Onyx.connect({
     key: ONYXKEYS.LOGIN_LIST,
-    callback: (val) => (loginList = val),
+    callback: (value) => (loginList = value),
 });
 
-function getChatType(report) {
-    return report ? report.chatType : '';
+function getChatType(report: OnyxEntry<Report>): ValueOf<typeof CONST.REPORT.CHAT_TYPE> | undefined {
+    return report?.chatType;
 }
 
-/**
- * @param {String} policyID
- * @returns {Object}
- */
-function getPolicy(policyID) {
+function getPolicy(policyID: string): OnyxEntry<Policy> | EmptyObject {
```

Comment:
```suggestion
function getPolicy(policyID: string): Policy | EmptyObject {
```
We are never returning null in this case